util/tempfile.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) The Test Authors
+// SPDX-License-Identifier: MPL-2.0
+
+package util
+
+import (
+	"io/fs"
+	"os"
+)
+
+type T interface {
+	TempDir() string
+	Helper()
+	Errorf(format string, args ...any)
+	Fatalf(format string, args ...any)
+	Cleanup(func())
+}
+
+type TempFileSettings struct {
+	data        []byte
+	mode        *fs.FileMode
+	namePattern string
+	path        *string
```

Comment:
I might have used 


```suggestion
	path        string
```

Checking path is not-empty would be enough, no?

Reply:
An empty path is meaningful, as it tells `os.CreateTemp` to use "the default directory for temporary files, as returned by [TempDir](https://pkg.go.dev/os#TempDir)." I forgot to document that behavior. I'll leave the pointer there, fix the doc comment on `Path`, and rename `Path` to `Dir` to match the argument in `os.CreateTemp`.

Reply:
Oh indeed. That's right. I faced this once. Then I forgot